src/ResourceManager/RecoveryServices/Commands.RecoveryServices/Common/RecoveryServicesCmdletBase.cs

```diff
@@ -68,36 +69,43 @@ public void HandleException(Exception ex)
             CloudException cloudException = ex as CloudException;
             if (cloudException != null)
             {
-                Error error = null;
+                ARMError error = null;
                 try
                 {
-                    using (Stream stream = new MemoryStream())
+                    if (cloudException.Message != null)
                     {
-                        if (cloudException.Message != null)
-                        {
-                            byte[] data = System.Text.Encoding.UTF8.GetBytes(cloudException.Message);
-                            stream.Write(data, 0, data.Length);
-                            stream.Position = 0;
+                        string originalMessage = cloudException.Error.OriginalMessage;
+                        error = JsonConvert.DeserializeObject<ARMError>(originalMessage);
 
-                            var deserializer = new DataContractSerializer(typeof(ErrorInException));
-                            error = (Error)deserializer.ReadObject(stream);
+                        string exceptionMessage = Properties.Resources.CloudExceptionDetails;
 
-                            throw new InvalidOperationException(
-                                string.Format(
-                                Properties.Resources.CloudExceptionDetails,
-                                error.Message,
-                                error.PossibleCauses,
-                                error.RecommendedAction,
-                                error.ClientRequestId));
+                        if (error.Error.Details != null)
+                        {
+                            foreach (ARMExceptionDetails detail in error.Error.Details)
+                            {
+                                exceptionMessage = exceptionMessage + string.Concat(
+                                    string.IsNullOrEmpty(detail.ErrorCode) ? "" : "\nErrorCode: " + detail.ErrorCode,
```

Comment:
Please use SrtingBuffer and then do AppendLine instead of adding \n in at each line. This code may run on non windows machines in the future and the output will be messed up.